pallets/oracle-data-collection/src/lib.rs

```diff
@@ -312,12 +352,18 @@ pub mod pallet {
 				})
 				.collect::<Result<Vec<_>, _>>()?;
 
-			T::AggregationProvider::aggregate(fed_values)
-				.ok_or(Error::<T>::KeyNotInCollection.into())
+			let (value, timestamp) = T::AggregationProvider::aggregate(fed_values)
+				.ok_or(Error::<T>::KeyNotInCollection)?;
+
+			Self::ensure_valid_timestamp(collection_id, timestamp)?;
+
+			Ok((value, timestamp))
```

Comment:
I'm aggregating both values and timestamps and from the aggregated timestamp, I'm checking that it's not outdated.

Another option could be to filter outdated timestamps and aggregate them later.

Reply:
Thinking a bit more. I think makes more sense the second option. To avoid aggregating over outdated values

Reply:
Option 3:

- **Final timestamp** is the median of the timestamps,
- If the median is not outdated then the **final value** is the median of the non-outdated Oracle values.

I see this option quite accurate, but the timestamp probably would be older than the average of used values.

@mustermeiszer

Reply:
Final solution (after Frederic sync):
- Filter outdated oracles until a limit N.
  - If the limit is overpassed, then `OracleOutdatedError`.
- value & timestamp median of the non-outdated oracles